imap_processing/tests/hit/test_hit_l1a.py

```diff
@@ -0,0 +1,106 @@
+import pathlib
+
+import pytest
+import xarray as xr
+
+from imap_processing import imap_module_directory
+from imap_processing.cdf.utils import write_cdf
+from imap_processing.hit.l0.data_classes.housekeeping import Housekeeping
+from imap_processing.hit.l1a import hit_l1a
+
+
+@pytest.fixture(scope="module")
+def packet_filepath():
+    """Set path to test data file"""
+
+    return imap_module_directory / "tests/hit/test_data/hskp_sample.ccsds"
+
+
+@pytest.fixture(scope="module")
+def unpacked_packets(packet_filepath):
+    """Read test data from file
+
+    Parameters
+    ----------
+    packet_filepath : str
+        Path to ccsds file
+
+    Returns
+    -------
+    sorted_packets : list[space_packet_parser.parser.Packet]
+        A sorted list of decommutated packets
+    """
+    packets = hit_l1a.decom_packets(packet_filepath)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    return sorted_packets
+
+
+def test_group_data(unpacked_packets):
+    """Group packet data classes by apid
+
+    Creates data classes for decommutated packets and
+    groups them by apid
+
+    Parameters
+    ----------
+    unpacked_packets : list[space_packet_parser.parser.Packet]
+        A sorted list of decommutated packets
+    """
+    grouped_data = hit_l1a.group_data(unpacked_packets)
+    assert len(grouped_data.keys()) == 1
+    assert all([isinstance(i, Housekeeping) for i in grouped_data[1251]])
+
+
+def test_create_datasets(unpacked_packets):
+    """Create xarray datasets
+
+    Creates a xarray dataset for each apid and stores
+    them in a dictionary
+
+    Parameters
+    ----------
+    unpacked_packets : list[space_packet_parser.parser.Packet]
+        A sorted list of decommutated packets
+    """
+    grouped_data = hit_l1a.group_data(unpacked_packets)
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+    datasets = hit_l1a.create_datasets(grouped_data, skip_keys=skip_keys)
+    assert len(datasets.keys()) == 1
+    assert isinstance(datasets[1251], xr.Dataset)
+
+
+def test_hit_l1a(packet_filepath):
+    """Create L1A CDF files
+
+    Creates a CDF file for each apid dataset and stores all
+    cdf file paths in a list
+
+    Parameters
+    ----------
+    packet_filepath : str
+        Path to ccsds file
+    """
+    packets = hit_l1a.decom_packets(packet_filepath)
+    sorted_packets = sorted(packets, key=lambda x: x.data["SHCOARSE"].derived_value)
+    grouped_data = hit_l1a.group_data(sorted_packets)
+    skip_keys = [
+        "shcoarse",
+        "ground_sw_version",
+        "packet_file_name",
+        "ccsds_header",
+        "leak_i_raw",
+    ]
+    datasets = hit_l1a.create_datasets(grouped_data, skip_keys=skip_keys)
+    cdf_filepaths = []
+    for dataset in datasets.values():
+        cdf_file = write_cdf(dataset)
+        cdf_filepaths.append(cdf_file)
```

Comment:
You're rewriting the `hit_l1a_data` code here I think, do you mean to call that function so you're testing it?
```suggestion
    cdf_filepaths = hit_l1a.hit_l1a_data(packet_filepath)
```

Reply:
This will also help with the `codecov` failures I think: https://app.codecov.io/gh/IMAP-Science-Operations-Center/imap_processing/pull/583/blob/imap_processing/hit/l1a/hit_l1a.py

Reply:
My last job did not have good testing practices, so I'm a novice at unit tests. Your feedback is much appreciated!